cwl/src/main/scala/cwl/CwlWomExpression.scala

```diff
@@ -122,14 +121,17 @@ object InitialWorkDirFileGeneratorExpression {
 
     implicit val caseExpression: Case.Aux[Expression, InitialWorkDirFileEvaluator] = {
       at { expression =>
-        (_, unmappedParameterContext, mappedInputValues, expressionLib) => {
+        (ioFunctionSet, unmappedParameterContext, _, expressionLib) => {
           // A single expression which must evaluate to an array of Files
           val expressionEvaluation = ExpressionEvaluator.eval(expression, unmappedParameterContext, expressionLib)
 
           expressionEvaluation flatMap {
             case array: WomArray if WomArrayType(WomSingleFileType).coercionDefined(array) =>
               WomArrayType(WomSingleFileType).coerceRawValue(array).toErrorOr
-            case file: WomFile => file.valid
+            case file: WomFile =>
+              // The file should be staged to the initial work dir using the base filename.
+              val baseFileName = file.value.substring(file.value.lastIndexOf('/') + 1)
```

Comment:
Hopefully Thibault's forthcoming IoFunctionSet changes can hide this from view 😄

Reply:
ToL: since you've been looking at it most recently, could you comment what this function, and the outer `Poly1`, is actually doing? I've looked at it a few times now and it's no less opaque with this change...

Reply:
I thought you wrote this?